src/pymor/parameters/functionals.py

```diff
@@ -45,11 +45,27 @@ def d_mu(self, parameter, index=0):
     def __call__(self, mu=None):
         return self.evaluate(mu)
 
+    def __add__(self, other):
+        if isinstance(other, Number) and other == 0:
+            return self
+        elif not isinstance(other, ParameterFunctional):
+            other = ConstantParameterFunctional(other)
+        return LincombParameterFunctional([self, other], [1., 1.])
```

Comment:
I found this a bit confusing to read. Maybe:
```suggestion
    def __add__(self, other):
        if isinstance(other, Number):
            if other == 0:
                return self
            other = ConstantParameterFunctional(other)
        if isinstance(other, ParameterFunctional):
            return LincombParameterFunctional([self, other], [1., 1.])
        else:
            return NotImplemented        
```